src/kernel/heap.zig

```diff
@@ -244,10 +244,16 @@ const FreeListAllocator = struct {
             return Allocator.Error.OutOfMemory;
         } else {
             // Shrinking
-            const size_diff = old_mem.len - real_size;
+            var size_diff = old_mem.len - real_size;
             // If shrinking would leave less space than required for a new header,
             // or if shrinking would make the buffer too small, don't shrink
-            if (size_diff < @sizeOf(Header) or real_size < @sizeOf(Header)) return Allocator.Error.OutOfMemory;
+            if (size_diff < @sizeOf(Header)) {
+                return old_mem.len;
```

Comment:
I believe that the `Allocator` API says that `OutOfMemory` should be returned if no resizing is happening, but that may be incorrect since there is some inconsistency in the documentation.

Reply:
I had to do this, bc I have a shrink that is lees than the size of a header, and there is nothing else I can do but do this.